app/src/main/java/com/murrayc/galaxyzoo/app/SubjectFragment.java

```diff
@@ -427,7 +438,63 @@ private void updateShareActionIntent() {
         //shareIntent.putExtra(Intent.EXTRA_STREAM, mUriImageStandard);
         //shareIntent.setType("image/*");
 
+        if (mUriStandardRemote!=null) {
+            GetImageBitmapAsyncTask getImageBitmapAsyncTask = new GetImageBitmapAsyncTask(){
+                @Override
+                protected void onPostExecute(Uri uri) {
+                    shareIntent.setType("image/*");
+                    shareIntent.putExtra(Intent.EXTRA_STREAM, uri);
+                    shareIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                    mShareActionProvider.setShareIntent(shareIntent);
+                }
+            };
+            getImageBitmapAsyncTask.execute(mUriStandardRemote);
+        }
+
 
         mShareActionProvider.setShareIntent(shareIntent);
```

Comment:
The second setShareIntent is set after the AsyncTask has started executing, but before it is completed. When the AsyncTask completes, the share intent is updated with the image type and content URI. I had left it there to provide a temporary intent on slower devices or in edge cases where the AsyncTask doesn't complete before the user decides to share, but it can indeed be removed. I will remove it in the next commit.

I would still leave the instantiation outside onPostExecute, since that leaves less main thread work to be done after the AsyncTask is completed.